R/kodebok-valider.R

```diff
@@ -36,69 +36,51 @@ kb_til_kanonisk_form = function(kb) {
   # Det kan vera at nokre ikkje-essensielle kolonnar manglar.
   # Då legg med dei til, med NA-verdiar eller eksempeldata,
   # alt etter kva som trengst.
-  kb = kb %>%
-    leggtil_std(skjema_id, "fiktiv_skjema_id")
-  kb = kb %>%
-    leggtil_std(skjemanamn, paste0("Skjemanamn for ", kb$skjema_id))
-  kb = kb %>%
-    leggtil_std(kategori, NA_character_)
-  kb = kb %>%
-    leggtil_std(innleiing, NA_character_)
-  kb = kb %>%
-    leggtil_std(variabeletikett, NA_character_)
-  kb = kb %>%
-    leggtil_std(forklaring, NA_character_)
-  kb = kb %>%
-    leggtil_std(manglande, NA_character_)
-  kb = kb %>%
-    leggtil_std(eining, NA_character_)
-  kb = kb %>%
-    leggtil_std(unik, "nei")
-  kb = kb %>%
-    leggtil_std(min, NA_real_)
-  kb = kb %>%
-    leggtil_std(maks, NA_real_)
-  kb = kb %>%
-    leggtil_std(min_rimeleg, NA_real_)
-  kb = kb %>%
-    leggtil_std(maks_rimeleg, NA_real_)
-  kb = kb %>%
-    leggtil_std(kommentar_rimeleg, NA_character_)
-  kb = kb %>%
-    leggtil_std(utrekningsformel, NA_character_)
-  kb = kb %>%
-    leggtil_std(logikk, NA_character_)
-  kb = kb %>%
+  kb = leggtil_std(kb, skjema_id, "fiktiv_skjema_id") |>
+    leggtil_std(skjemanamn, paste0("Skjemanamn for ", kb$skjema_id)) |>
```

Comment:
Vil denne nødvendigvis fungera? Eg veit ikkje kva `leggtil_std()` gjer, men viss denne legg til ekstra rader eller endrar rekkjefølgja på radene, vil det føra til problem, sidan `kb`-objektet er endra. (Det ville ha fungert med den gamle koden, sidan `kb` der vart oppdatert undervegs.)

Reply:
leggtil_std() legger til manglende standard-kolonner fra kanonisk kodebok hvis de ikke allerede finnes i inndata. Siden disse legges til som siste kolonne og i samme rekkefølge som tidligere blir det tilsvarende resultat som med den gamle koden.

Reply:
> Vil denne nødvendigvis fungera? Eg veit ikkje kva `leggtil_std()` gjer, men viss denne legg til ekstra rader eller endrar rekkjefølgja på radene, vil det føra til problem, sidan `kb`-objektet er endra. (Det ville ha fungert med den gamle koden, sidan `kb` der vart oppdatert undervegs.)

Ja, ser no at nett denne linja, som brukar `kb` som argument, ikkje vil fungera for kodebøker som manglar `skjema_id` i utgangspunktet (linja rett før legg til `skjema_id = "fiktiv_skjema_id"` dersom `skjema_id` manglar).

Kva er beste alternativ? Berre la alle linjene overskriva `kb` slik det stod før kommentaren min? Eventuelt la den fyrste linja overskriva `kb`, og så bruka røyr på dei resterande som ikkje brukar `kb` som argument til funksjonen?

Reply:
Eg synest det er best at ein lèt den første linja overskriva `kb` og så bruka røyr på dei resterande som ikkje brukar `kb`. Men må stå ein kommentar som forklarar, slik at ikkje nokon seinare kjem og «optimaliserar» dette vekk.

Reply:
Aha, da skjønner jeg. Har lagt inn ny commit hvor `kb` overskrives for `skjema_id`. Takk for innspill.